### ISSUE TITLE
Backpropogation

### ISSUE THREAD

**Issue by hdmGOAT** [hdmGOAT/trustformer#8]
- [ ] Gradient buffers
- [ ] Saved activiations
- [ ] Chain rule
- [ ] LayerNorm, Attention, Softmax Backward
- [ ] Embedding gradient accumulation
